src/nsls2api/services/slack_service.py

```diff
@@ -0,0 +1,223 @@
+    except SlackApiError as error:
+        logger.exception(error)
+        return []
+    return response.data["members"]
+
+
+def add_bot_to_channel(channel_id: str):
+    """
+    Adds the bot to the specified channel.
+
+    Args:
+        channel_id (str): The ID of the channel to add the bot to.
+
+    Raises:
+        Exception: If an error occurs while adding the bot to the channel.
+
+    Returns:
+        None
+    """
+    bot = get_bot_details()
+    client = WebClient(token=settings.superadmin_slack_user_token)
+    logger.info(f"Inviting {bot.username} ({bot.user_id}) to channel {channel_id}.")
+    try:
+        response = client.admin_conversations_invite(
+            channel_id=channel_id, user_ids=[bot.user_id]
+        )
+        logger.info(response)
+    except SlackApiError as error:
+        if error.response["error"] == "failed_for_some_users":
+            channel_members = get_channel_members(channel_id)
+            if "failed_user_ids" in error.response:
+                if (bot.user_id in error.response["failed_user_ids"]) and (
+                    bot.user_id in channel_members
+                ):
+                    logger.info(f"{bot.username} is already in channel {channel_id}.")
+                else:
+                    logger.error(
+                        f"Failed to add bot {bot.username} to channel {channel_id}."
+                    )
+        else:
+            logger.exception(error)
+            raise Exception(error) from error
+
+
+async def create_channel(
+    name: str, is_private: bool = True, description: str = None
+) -> str | None:
+    """
+    Creates a new Slack channel with the given name, privacy settings, and description.
+
+    Args:
+        name (str): The name of the channel to be created.
+        is_private (bool, optional): Whether the channel should be private. Defaults to True.
+        description (str, optional): The description of the channel. Defaults to None.
+
+    Returns:
+        str | None: The ID of the created channel if successful, None otherwise.
+    """
+    super_client = WebClient(token=settings.superadmin_slack_user_token)
+
+    # Does the channel already exist?
+    channel_id = channel_id_from_name(name)
+
+    if channel_id:
+        logger.info(f"Found existing channel called {name}.")
+    else:
+        try:
+            logger.info(f"Trying to create channel called '{name}'.")
+            response = super_client.admin_conversations_create(
+                name=name,
+                is_private=is_private,
+                description=description,
+                team_id=settings.nsls2_workspace_team_id,
+            )
+            logger.info(f"Response from creating slack channel: {response}")
+
+        except SlackApiError as e:
+            logger.exception(f"Error creating channel: {e}")
+            return None
+
+    # Now lets add our 'bot' to the channel
+    add_bot_to_channel(channel_id)
```

Comment:
I put it there because some of the other functions then rely on the bot being in the channel - so I decided to just add the bot to all channels we create.